nemo/collections/asr/modules/hybrid_autoregressive_transducer.py

```diff
@@ -0,0 +1,255 @@
+            all on sub-batches, then collates results to be exactly equal to results from the entire batch.
+
+            When this flag is set, prior to calling forward, the fields `loss` and `wer` (either one) *must*
+            be set using the `HATJoint.set_loss()` or `HATJoint.set_wer()` methods.
+
+            Further, when this flag is set, the following argument `fused_batch_size` *must* be provided
+            as a non negative integer. This value refers to the size of the sub-batch.
+
+            When the flag is set, the input and output signature of `forward()` of this method changes.
+            Input - in addition to `encoder_outputs` (mandatory argument), the following arguments can be provided.
+                - decoder_outputs (optional). Required if loss computation is required.
+                - encoder_lengths (required)
+                - transcripts (optional). Required for wer calculation.
+                - transcript_lengths (optional). Required for wer calculation.
+                - compute_wer (bool, default false). Whether to compute WER or not for the fused batch.
+
+            Output - instead of the usual `joint` log prob tensor, the following results can be returned.
+                - loss (optional). Returned if decoder_outputs, transcripts and transript_lengths are not None.
+                - wer_numerator + wer_denominator (optional). Returned if transcripts, transcripts_lengths are provided
+                    and compute_wer is set.
+
+        fused_batch_size: Optional int, required if `fuse_loss_wer` flag is set. Determines the size of the
+            sub-batches. Should be any value below the actual batch size per GPU.
+    """
+
+    def __init__(
+        self,
+        jointnet: Dict[str, Any],
+        num_classes: int,
+        num_extra_outputs: int = 0,
+        vocabulary: Optional[List] = None,
+        log_softmax: Optional[bool] = None,
+        preserve_memory: bool = False,
+        fuse_loss_wer: bool = False,
+        fused_batch_size: Optional[int] = None,
+        experimental_fuse_loss_wer: Any = None,
+    ):
+        super().__init__(
+            jointnet=jointnet,
+            num_classes=num_classes,
+            num_extra_outputs=num_extra_outputs,
+            vocabulary=vocabulary,
+            log_softmax=log_softmax,
+            preserve_memory=preserve_memory,
+            fuse_loss_wer=fuse_loss_wer,
+            fused_batch_size=fused_batch_size,
+            experimental_fuse_loss_wer=experimental_fuse_loss_wer,
+        )
+
+        self.pred, self.enc, self.joint_net, self.blank_pred = self._joint_hat_net_modules(
+            num_classes=self._vocab_size,  # non blank symbol
+            pred_n_hidden=self.pred_hidden,
+            enc_n_hidden=self.encoder_hidden,
+            joint_n_hidden=self.joint_hidden,
+            activation=self.activation,
+            dropout=jointnet.get('dropout', 0.0),
+        )
```

Comment:
Add a property (settable) to this class that sets whether to return_hat_ilm (default to false)